### ISSUE TITLE
1.12 Installer fails on Windows 10 attempting a install over existing 1.10 instance.

### ISSUE THREAD

**Issue by awslovato** [puppetlabs/pdk#740]
**Describe the bug**
Using the official .msi from https://puppet.com/download-puppet-development-kit to upgrade from 1.10 to 1.12 fails with an error. It removed 1.10 totally so I don't have that any more either.

**To Reproduce**
Impossible to reproduce now that 1.10 is no longer on my host, but using the msi to attempt an install gets to the same spot and then fails with the same error. I have admin privileges on my machine.  I did reboot in between 2 of the 3 install attempts.

**Expected behavior**
1.10 to be uninstalled, 1.12 to install without issues. If there was an issue, a true rollback to 1.10
would be helfpul.

**Additional context**
- native MSI
- no version (previously 1.10)
- windows 10

Attached Msi log to help diagnose.  Thanks in advance.
[MSIc7545.zip](https://github.com/puppetlabs/pdk/files/3557455/MSIc7545.zip)

James

**Comment by glennsarti** [comment]
Thanks for the bug report @awslovato .  And also huge thanks for the MSI log file.

Looks like it failed here
```
MSI (s) (0C:60) [15:27:53:758]: Creating MSIHANDLE (11) of type 790531 for thread 21600
1: Creating process level environment variables... 
MSI (s) (0C:60) [15:27:53:759]: Closing MSIHANDLE (11) of type 790531 for thread 21600
MSI (s) (0C:60) [15:27:53:761]: Creating MSIHANDLE (12) of type 790531 for thread 21600
1: Executing Command : C:\Windows\system32\cmd.exe /C C:\PROGRA~1\Puppet Labs\DevelopmentKit\private\ruby\2.4.5\bin\ruby.exe -S -- C:\PROGRA~1\Puppet Labs\DevelopmentKit\share\install-tarballs\extract_all.rb 
MSI (s) (0C:60) [15:27:53:762]: Closing MSIHANDLE (12) of type 790531 for thread 21600
MSI (s) (0C:60) [15:27:53:816]: Creating MSIHANDLE (13) of type 790531 for thread 21600
1: 'C:\PROGRA~1\Puppet' is not recognized as an internal or external command, 
MSI (s) (0C:60) [15:27:53:817]: Closing MSIHANDLE (13) of type 790531 for thread 21600
MSI (s) (0C:60) [15:27:53:817]: Creating MSIHANDLE (14) of type 790531 for thread 21600
1: operable program or batch file. 
MSI (s) (0C:60) [15:27:53:818]: Closing MSIHANDLE (14) of type 790531 for thread 21600
MSI (s) (0C:60) [15:27:53:818]: Creating MSIHANDLE (15) of type 790531 for thread 21600
1: Execution Failed With Code: 1 
MSI (s) (0C!38) [15:27:55:416]: Note: 1: 2205 2:  3: Error 
MSI (s) (0C!38) [15:27:55:416]: Note: 1: 2228 2:  3: Error 4: SELECT `Message` FROM `Error` WHERE `Error` = 1709 
MSI (s) (0C!38) [15:27:55:416]: Product: Puppet Development Kit -- 1: Execution Failed With Code: 1 
```

This looks exactly like the same problem in #721  Note that the path is `...Puppet Labs...` with a space.  Good news is, this should already be fixed in 1.13.0!

The rollback to 1.10 is unfortunately expected.  This is a [Major Upgrade](https://docs.microsoft.com/en-us/windows/win32/msi/major-upgrades) in the MSI sense and it uninstalls the old one first, which means you can't rollback to the previous one.

I'll leave this issue open for a couple of weeks and give you a chance to try out 1.13 and make sure it works for you and comment here with your results.

**Comment by awslovato** [comment]
> 
> 
> Thanks for the bug report @awslovato . And also huge thanks for the MSI log file.
> 
> Looks like it failed here
> 
> ```
> MSI (s) (0C:60) [15:27:53:758]: Creating MSIHANDLE (11) of type 790531 for thread 21600
> 1: Creating process level environment variables... 
> MSI (s) (0C:60) [15:27:53:759]: Closing MSIHANDLE (11) of type 790531 for thread 21600
> MSI (s) (0C:60) [15:27:53:761]: Creating MSIHANDLE (12) of type 790531 for thread 21600
> 1: Executing Command : C:\Windows\system32\cmd.exe /C C:\PROGRA~1\Puppet Labs\DevelopmentKit\private\ruby\2.4.5\bin\ruby.exe -S -- C:\PROGRA~1\Puppet Labs\DevelopmentKit\share\install-tarballs\extract_all.rb 
> MSI (s) (0C:60) [15:27:53:762]: Closing MSIHANDLE (12) of type 790531 for thread 21600
> MSI (s) (0C:60) [15:27:53:816]: Creating MSIHANDLE (13) of type 790531 for thread 21600
> 1: 'C:\PROGRA~1\Puppet' is not recognized as an internal or external command, 
> MSI (s) (0C:60) [15:27:53:817]: Closing MSIHANDLE (13) of type 790531 for thread 21600
> MSI (s) (0C:60) [15:27:53:817]: Creating MSIHANDLE (14) of type 790531 for thread 21600
> 1: operable program or batch file. 
> MSI (s) (0C:60) [15:27:53:818]: Closing MSIHANDLE (14) of type 790531 for thread 21600
> MSI (s) (0C:60) [15:27:53:818]: Creating MSIHANDLE (15) of type 790531 for thread 21600
> 1: Execution Failed With Code: 1 
> MSI (s) (0C!38) [15:27:55:416]: Note: 1: 2205 2:  3: Error 
> MSI (s) (0C!38) [15:27:55:416]: Note: 1: 2228 2:  3: Error 4: SELECT `Message` FROM `Error` WHERE `Error` = 1709 
> MSI (s) (0C!38) [15:27:55:416]: Product: Puppet Development Kit -- 1: Execution Failed With Code: 1 
> ```
> 
> This looks exactly like the same problem in #721 Note that the path is `...Puppet Labs...` with a space. Good news is, this should already be fixed in 1.13.0!
> 
> The rollback to 1.10 is unfortunately expected. This is a [Major Upgrade](https://docs.microsoft.com/en-us/windows/win32/msi/major-upgrades) in the MSI sense and it uninstalls the old one first, which means you can't rollback to the previous one.
> 
> I'll leave this issue open for a couple of weeks and give you a chance to try out 1.13 and make sure it works for you and comment here with your results.

Hi @glennsarti ,
No problem about the rollback, I just didn't have a copy of 1.10 laying around. I did install 1.13 and it went without a hitch. As this was a dupe of #721 I can close this as the issue was with the installation, and nothing else.

Thanks for your assistance,
James
